Orange/widgets/unsupervised/owtsne.py

```diff
@@ -215,12 +226,25 @@ def stop(self):
     def resume(self):
         self.__set_update_loop(self.tsne_iterator)
 
+    def set_data(self, data: Table):
+        super().set_data(data)
+
+        if data is not None:
+            # PCA doesn't support normalization on sparse data, as this would
+            # require centering and normalizing the matrix
+            self.normalize_cbx.setDisabled(data.is_sparse())
+
     def pca_preprocessing(self):
-        if self.pca_data is not None and \
-                self.pca_data.X.shape[1] == self.pca_components:
```

Comment:
I removed this check because I changed line 146 to invalidate the PCA projection, therefore it will be set to `None` whenever the number of components changes.